{executorch][llama] support mqa

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3080

This diff adds support for multi query attention for sdpa with kv cache

Differential Revision: [D56228316](https://our.internmc.facebook.com/intern/diff/D56228316/)

kimishpatel committed:
{executorch][llama] support mqa

This diff adds support for multi query attention for sdpa with kv cache

Differential Revision: [D56228316](https://our.internmc.facebook.com/intern/diff/D56228316/)

[ghstack-poisoned]

mikekgfb reviewed (approved):
Please ensure all tests pass before landing!

larryliu0820 commented on examples/models/llama2/custom_ops/TARGETS:
```diff
@@ -1,8 +1,22 @@
 # Any targets that should be shared between fbcode and xplat must be defined in
 # targets.bzl. This file can contain fbcode-only targets.
 
+load("@fbcode_macros//build_defs:python_unittest.bzl", "python_unittest")
```
Use `runtime`? You can do

```
runtime.python_unittest(
...
)
```

kimishpatel replied:
oh this was due to arc lint. let me fix.

kimishpatel committed:
Update on "{executorch][llama] support mqa"


This diff adds support for multi query attention for sdpa with kv cache

Differential Revision: [D56228316](https://our.internmc.facebook.com/intern/diff/D56228316/)

[ghstack-poisoned]

kimishpatel commented:
bunch of pre-existing failures with re2.h that @larryliu0820  is fixing
